Nixos migration

al12gamer opened the pull request:
This will probably be a long-running pull request to eventually move all extra yay or pacman packages I added to my Deck previously, to the nix package manager.

Per [this comment here](https://github.com/NixOS/nix/issues/7173#issuecomment-1604717502), it has been rumored that the Steam Deck will include a `/nix` directory by default in the future. In learning about NixOS, I want to move more immutable installs to use nix packages.

al12gamer committed:
Update README.md

updated for clarity, add nix packages

al12gamer committed:
Update deckscript.sh

update to include nix, remove unnecessary pacman packages

al12gamer committed:
add blank nix configuration

al12gamer commented:
Unable to run the nix install script currently, as `/nix` is not available on the Steam Deck currently, per this output:
```
~~> Setting up the basic directory structure

---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo install -dv -m 0755 /nix /nix/var /nix/var/log /nix/var/log/nix /nix/var/log/nix/drvs /nix/var/nix /nix/var/nix/db /nix/var/nix/gcroots /nix/var/nix/profiles /nix/var/nix/temproots /nix/var/nix/userpool /nix/var/nix/daemon-socket /nix/var/nix/gcroots/per-user /nix/var/nix/profiles/per-user

to make the basic directory structure of Nix (part 1)

install: cannot change permissions of ‘/nix’: No such file or directory
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system
install: cannot create directory ‘/nix’: Read-only file system

---- oh no! --------------------------------------------------------------------
Oh no, something went wrong. If you can take all the output and open
an issue, we'd love to fix the problem so nobody else has this issue.

:(

We'd love to help if you need it.

You can open an issue at
https://github.com/NixOS/nix/issues/new?labels=installer&template=installer.md

Or get in touch with the community: https://nixos.org/community
```

al12gamer commented:
[This](https://nixos.org/download.html#nix-install-linux) is the script that  I ran to install on the Deck currently. We may need to wait until the `/nix` directly is included by default on Deck. Should it become available in the Preview SteamOS channel in the future, I may switch to test it out.

al12gamer commented:
The goal here is to move all the pacman/yay packages to Nix packages, so as to only have nix and flatpak as the packages the Deck install is using going forward for not-installed-by-default software.

al12gamer commented:
...will likely update the config file when I can generate a base config on the Deck, testing any suggestions the Valve employee may have

miki223 commented:
oh interesting fellow nix enjoyer 
i assume you know about this project
https://github.com/Jovian-Experiments/Jovian-NixOS

al12gamer commented:
Oh I do now! That's pretty fantastic! This repo is mainly for setting up the deck the way I would use it with SteamOS for now, but migrating everything to nix packages would simplify quite a lot.

al12gamer commented:
Looks like a ChimeraOS dev mentioned they may add nix here
https://github.com/NixOS/nix/issues/7173#issuecomment-1614577273
Still waiting on further notes from the Valve developers.

al12gamer commented:
This could also likely fix #11 as it seems airshipper is available as a [Nix package here](https://search.nixos.org/packages?channel=23.05&show=airshipper&from=0&size=50&sort=relevance&type=packages&query=airshipper)

al12gamer commented:
Looks like StamOS 3.5 will include the nix directory. Waiting for that release before making further progress moving this to nix packages.

al12gamer commented:
This may also prove useful https://github.com/Jovian-Experiments/Jovian-NixOS/

al12gamer commented:
Once the Nix directory is officially added to SteamOS, this will likely be the final PR to this repo

al12gamer commented:
[Per a post on Reddit](https://www.reddit.com/r/SteamDeck/comments/16qq1hn/question_for_those_on_35_preview_is_the_nix/), it looks to be that SteamOS 3.6 has the `/nix` directory enabled by default, so when that hits I'll be able to migrate these scripts.

al12gamer committed:
Update deckscript.sh

Migrated Firefox, btop, Brave, VSCodium, jotta-cli and Gnome Boxes to Nix Packages. Added Nixpkg installer for single-user. Removed steps and packages requiring Arch packages or disabling read-only mode. Updated notes.

al12gamer committed:
Update deckscript.sh

al12gamer commented:
Began package migration now that the nix directory is going to be available in future releases. Waiting to complete pull request until the SteamOS preview becomes at least a Beta.

al12gamer commented:
(didn't mean to close it)

al12gamer committed:
Update SoH.sh

updated opening of github page instead of heading to discord- looks like development is now done on github instead

al12gamer committed:
Update deckscript.sh

Make the retroarch download all cores script manual

al12gamer committed:
Delete configuration.nix

removing configuration for now, as this will be created when running the Nix installer or when running the Jovian nix project for Deck

al12gamer committed:
Update deckscript.sh

correct jotta-cli package name and edit SoH wording

al12gamer commented:
I might add an EmuDeck download too

al12gamer committed:
Update deckscript.sh

download emudeck installer

al12gamer committed:
Update README.md

No more pacman packages, so removed mention of it. Corrected what packages are installed.

al12gamer committed:
Update README.md

al12gamer commented:
Adding back syncthingy as I use it across four machines

al12gamer committed:
Update deckscript.sh

Add SyncThingy flatpak back in for syncing saves that don't have cloud options

al12gamer committed:
Update README.md

add SyncThingy

al12gamer committed:
Update README.md

correcting - no longer need to disable steamos read only

al12gamer commented:
SteamOS 3.5.5 hit the Stable channel today or yesterday. I'll have to dock the Deck and test to see if the Nix ENV installer works when I get some time.

al12gamer commented:
Would be interesting to see if I can automate #16 through nix packages......if not, perhaps the flatpak will do it fine.

al12gamer commented:
Am currently receiving this error when trying to install the btop package, after running the multi-user installer for a Nix package environment on the Deck. That said, the `/nix` directory is definitely there.
```
(1)(deck@gabegear ~)$ nix-env -iA nixpkgs.btop
error:
       … while evaluating a branch condition

         at /nix/store/0z1350wf56r5yydpqv89q9g3a26i8565-nixpkgs/nixpkgs/pkgs/stdenv/booter.nix:99:7:

           98|     thisStage =
           99|       if args.__raw or false
             |       ^
          100|       then args'

       … in the right operand of the update (//) operator

         at /nix/store/0z1350wf56r5yydpqv89q9g3a26i8565-nixpkgs/nixpkgs/pkgs/stdenv/booter.nix:84:7:

           83|       { allowCustomOverrides = index == 1; }
           84|       // (stageFun prevStage))
             |       ^
           85|     (lib.lists.reverseList stageFuns);

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: cannot connect to socket at '/nix/var/nix/daemon-socket/socket': Connection refused

```
Here's the directory:
```
(deck@gabegear ~)$ ls /etc | grep nix
bash.bashrc.backup-before-nix
nix

```

al12gamer committed:
Update README.md

correct SteamOS version

al12gamer commented:
Trying what was noted in [this github issue comment ](https://github.com/NixOS/nix/issues/7173#issuecomment-1826879713)to get the single-user of nix-env installed.
Note that I removed the prior contents of `/nix` but not the folder, so anything from prior attempts were removed.

al12gamer commented:
Single-user install is working
https://asciinema.org/a/623454

al12gamer committed:
Update deckscript.sh

al12gamer committed:
Update README.md

al12gamer committed:
Update deckscript.sh

add fastfetch nix package

al12gamer commented:
added fastfetch from the unstable nix packages

al12gamer commented on SoH.sh:
needs correction as a discord invite is no longer required

al12gamer committed:
Update SoH.sh

invites no longer needed

al12gamer reviewed (commented):
looks good
